File: api/src/test/java/org/pmiops/workbench/exceptions/ExceptionUtilsTest.java
package org.pmiops.workbench.exceptions;

import static org.junit.jupiter.api.Assertions.assertThrows;

import java.net.SocketTimeoutException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class ExceptionUtilsTest {

  @Test
  public void convertNotebookException() throws Exception {
    org.pmiops.workbench.notebooks.ApiException cause =
        new org.pmiops.workbench.notebooks.ApiException(new SocketTimeoutException());
    assertThrows(
        GatewayTimeoutException.class, () -> ExceptionUtils.convertNotebookException(cause));
  }
}

File: api/src/test/java/org/pmiops/workbench/exceptions/ExceptionUtilsTest.kt
package org.pmiops.workbench.exceptions

import java.net.SocketTimeoutException
import org.junit.Test
import org.pmiops.workbench.notebooks.ApiException

class ExceptionUtilsTest {

    @Test(expected = GatewayTimeoutException::class)
    @Throws(Exception::class)
    fun convertNotebookException() {
        val cause = ApiException(SocketTimeoutException())
        ExceptionUtils.convertNotebookException(cause)
    }
}
